### ISSUE TITLE
Data Storage Folder File Transfer not a background task. 

### ISSUE THREAD

**Issue by ThisIsPaulDaily** [osmandapp/OsmAnd#13867]
### Description
I recently changed SD Cards in my phone and while importing backup settings I discovered that I could change the Data Storage Folder. However, when I switched from internal storage to my SD card it immediately started copying in the foreground. 
I tapped off of the copying and the app needed a restart. Upon restarting the app, it was pulling the maps from the new location, but since almost nothing had transferred, I had no existing maps. 

I then had to restart the app after changing the storage location back to internal storage (external 1?) where my maps worked again. 
Now I'm sitting for a very long time as it transfers all of the map files I have downloaded to my SD card.  I can't turn the screen off or disturb it while transferring or it stops the transfer. 

### How to reproduce?
Change Data storage folder from internal to external. 
Observe the "Copying OsmAnd data files..." dialog and progress bar. 
Tap elsewhere on the screen. 
Allow the app to restart. 
Observe maps missing. 
Repeat and switch back to internal storage.
Observe maps present. 
Repeat and wait.

### Desired Behavior
Ideally this would be a notification and a background transfer. I have like 17+ gigs of offline maps and data downloaded. I can't (Too afraid to try) even turn the screen brightness down to save battery right now. If someone called me I'd have to repeat the process. 

### Your Environment
OsmAnd Version: Version 4.1.11 (421) from F-Droid
Android/iOS version: 11 
Device model: Moto G7+ running Lineage OS Lake

**Maps used (offline):**   
If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs and its edition date.

**Comment by ThisIsPaulDaily** [comment]
I'm going to go ahead and add that it _feels_ that the transfer of files is artificially slowed down by the app dialogs progress bar. 
These files are relatively small and my SD card speed is good with sustained writes at 30+Mb/s (U3 V30).
 Since most progress bars don't mean anything maybe consider revising how it shows progress.  I know I have like 120K+ Jpeg tile satellite images downloaded that it's transferring pretty fast (each is under a second), but it still seems slow. Around 40- 45 seconds for "Slope Us Wisconsin northamerica.sqlitedb" to transfer. 

I get that this is something that should only need to be done once maybe by a user, but the experience doing it is really bad. 

_Rambling Anecdote Below_ 
I worked as QA for a software company once that had a bug that left a ~200mb file on your computer remaining when the uninstaller finished. Nobody noticed it in the field since our customers don't uninstall the software. My ticket was closed as a duplicate to one that had been made 11 years prior. The reason I mention this is that while something might not seem like a big deal since it is an edge case, it can be when there's one user who does that edge case a lot (such as the intern who verified the nightly builds).

**Comment by kevinmcmurtrie** [comment]
You're probably suffering from Google's attack on external storage.  They recently started requiring Play Store apps to use APIs that are 5x to 10000x slower.

https://github.com/osmandapp/OsmAnd/issues/13292 and lots of other tickets.

You might be able to disable forced scoped storage in LineageOS.  If you can, move your maps out of the Android scoped storage directory and it will be full speed again.

**Comment by vshcherb** [comment]
Merged with other discussions.

**Comment by ThisIsPaulDaily** [comment]
Thank you, I did a cursory search around before creating this ticket and missed the larger picture and issue that you tagged. It does appear that google is at fault for this. 

It ended up taking more than 2.5 hours to transfer all the files, and during the process I wasn't sure if it was doing anything at all since I had plugged my phone into my computer to charge and could see the storage volume size not changing. 

My old SD card was suffering from bitrot/data corruption so I needed to update, but now that I switched cards several apps are giving me new prompts for storage locations and suffering from these issues. Maybe it's becoming more apparent to me and was there the whole time. 

Keep up the good work.
